stan/math/gpu/add.hpp

```diff
@@ -21,7 +21,7 @@ namespace math {
  * input matrices do not have matching dimensions
  *
  */
-inline matrix_gpu add(const matrix_gpu& A, const matrix_gpu& B) {
+inline auto add(const matrix_gpu& A, const matrix_gpu& B) {
```

Comment:
This is fine but in general I think it's fine to leave return types as non-auto especially here where it's not a lot of typing to specify. Types are also documentation even if the compiler can infer them.